src/RuleSet/DeclarationBlock.php

```diff
@@ -64,8 +65,21 @@ public static function parse(ParserState $parserState, ?CSSList $list = null): ?
                             }
                         }
                         break;
-                    case ',':
+                    case '(':
+                        if (!\is_string($stringWrapperCharacter)) {
+                            ++$functionNestingLevel;
+                        }
+                        break;
+                    case ')':
                         if (!\is_string($stringWrapperCharacter)) {
+                            if ($functionNestingLevel <= 0) {
+                                throw new UnexpectedTokenException('anything but', ')');
```

Comment:
Is there any way we can test for this case?

Reply:
Yes, we can.  And we can also reject unclosed parentheses.  Have committed a change to address this.

There is an issue that the first character of the selector is consumed without inspection, but that is beyond the scope of this PR.